### PR TITLE
Change profile link to use a click handler.

### PR DESCRIPTION
**Pull request by kadams54**
Fixes https://github.com/juju/juju-gui/issues/1528

### CODE REVIEW

**Comment by kadams54** [comment]
## QA

Verify that the link in the breadcrumb to the user profile still works. Also test in a real env and make sure things work, ideally in one where the absolute URL would break (see the fixed issue for @frankban's setup).

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4803/
Test PASSed.

**Comment by hatched** [comment]
Can you add a test for actually clicking on the link by calling onClick directly

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4804/
Test FAILed.

**Comment by urosj** [comment]
QA 🆗

File: jujugui/static/gui/src/app/components/header-breadcrumb/test-header-breadcrumb.js [hunk 1]
```diff
@@ -169,4 +172,42 @@ describe('HeaderBreadcrumb', () => {
...
+        uncommittedChanges={false} />, true);
+    var instance = component.getMountedInstance();
+    var output = component.getRenderOutput();
+    instance._handleProfileClick({
+      preventDefault: sinon.stub()
+    });
+    assert.equal(changeState.called, true,
+                 'changeState was not called');
+    var state = changeState.args[0][0];
+    assert.deepEqual(state.sectionC.component, 'profile',
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/header-breadcrumb/test-header-breadcrumb.js]: If you want to use deep equal you should probably check that the metadata is also set to null

**Comment by hatched** [comment]
:+1: With trivial

File: jujugui/static/gui/src/app/components/header-breadcrumb/test-header-breadcrumb.js (hunk 1 above)
Comment (kadams54) [inline jujugui/static/gui/src/app/components/header-breadcrumb/test-header-breadcrumb.js]: Yeah, whoops, my bad. I originally compared the entire state object but then decided that the component was the primary thing we wanted to assert on. That way if we add other information to the state object in the future the test doesn't necessarily break.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4806/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4807/
Test PASSed.

**Comment by kadams54** [comment]
:shipit: Thanks!

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4808/
Test PASSed.

**Pull request merged**